### ISSUE TITLE
Log in - E-mail input field not focused when scroll down and click Create A New Account/Log in

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#32508]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**1.4.8-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Click outside е-mail input field so "Please enter an email or phone number' is presented
3. Scroll down to the bottom of the page
4. Click on Create A New Account or Log in

## Expected Result:
Е-mail input field should be focused

## Actual Result:
E-mail input field not focused when scroll down and click Create A New Account/Log in while the message 'Please enter an email or phone number' is presented

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/c9667c72-ee8a-49d7-9dae-28149e867d42

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by shubham1206agra** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue

Log in - E-mail input field not focused when scroll down and click Create A New Account/Log in

### What is the root cause of that problem?

This is due to missing focus call and we just scroll to top in this case

### What changes do you think we should make in order to solve the problem?

Add explicit focus mechanism

### What alternative solutions did you explore? (Optional)

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- Log in - E-mail input field not focused when scroll down and click Create A New Account/Log in

### What is the root cause of that problem?
- Currently, once we click on "Create A New Account" or "Log In", it just scroll to the top and clear all the session data as bellow:
https://github.com/Expensify/App/blob/0a8ad473e71c18adb3d299f0562057177de09e2a/src/pages/signin/SignInPageLayout/Footer.js#L30-L35
### What changes do you think we should make in order to solve the problem?
- First create the ```LoginFormInputManager.ts```:
```
import React from 'react';
import {TextInput} from 'react-native';

const loginFormInputRef = React.createRef<TextInput>();
const setFormError = () => {};

export default {loginFormInputRef, setFormError};
```
- Then, update:
https://github.com/Expensify/App/blob/0a8ad473e71c18adb3d299f0562057177de09e2a/src/pages/signin/LoginForm/BaseLoginForm.js#L251
to:
```
                    ref={(el) => {
                        input.current = el;
                        LoginFormInputManager.loginFormInputRef.current = el;
                    }}
```
- Finally, update:
https://github.com/Expensify/App/blob/0a8ad473e71c18adb3d299f0562057177de09e2a/src/pages/signin/SignInPageLayout/Footer.js#L30-L35
to:
````diff
const navigateHome = (scrollPageToTop) => {
    scrollPageToTop();

    // We need to clear sign in data in case the user is already in the ValidateCodeForm or PasswordForm pages
    Session.clearSignInData();
+  LoginFormInputManager.loginFormInputRef.current.focus();
};
````
### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by shubham1206agra** [comment]
@sobitneupane FYI - I didn't put the whole implementation code in my proposal, as this is a pretty straightforward solution to solve

**Comment by DylanDylann** [comment]
@shubham1206agra yeah. I think it depends on review to choose the solution for this issue

**Comment by sobitneupane** [comment]
@shubham1206agra Thanks for your [proposal](https://github.com/Expensify/App/issues/32508#issuecomment-1841459340).

Could you please add more details to your proposal? Please do include permalinks to point out where the issue exists. Where will you make the suggested changes? And more details on the solution will be appreciated.

**Comment by shubham1206agra** [comment]
I will. Thanks @sobitneupane

**Comment by shubham1206agra** [comment]
@sobitneupane Updated

**Comment by garrettmknight** [comment]
@sobitneupane can you take a look at the updated proposal when you get a chance?

**Comment by DylanDylann** [comment]
@sobitneupane What do you think about my [proposal](https://github.com/Expensify/App/issues/32508#issuecomment-1846757136)?

**Comment by garrettmknight** [comment]
@sobitneupane any thoughts?

**Comment by sobitneupane** [comment]
Thanks for the proposal @shubham1206agra and @DylanDylann 

Both of the proposal solves the issue. 

Though, [proposal](https://github.com/Expensify/App/issues/32508#issuecomment-1846757136) from @DylanDylann is quite simpler to implement, I think the [proposal](https://github.com/Expensify/App/issues/32508#issuecomment-1841459340) proposed by @shubham1206agra is the general practice of dealing with such cases. So, I am recommending @shubham1206agra's proposal.

🎀 👀 🎀 C+ reviewed

**Comment by garrettmknight** [comment]
@shubham1206agra you need any help to get the PR for this one up?

**Comment by garrettmknight** [comment]
@shubham1206agra you need any help to get the PR for this one up?

**Comment by shubham1206agra** [comment]
> @shubham1206agra you need any help to get the PR for this one up?

Sorry for the delay. I will open a PR shortly

**Comment by shubham1206agra** [comment]
I will publish a PR today

**Comment by tylerkaraszewski** [comment]
@shubham1206agra ?

**Comment by shubham1206agra** [comment]
Extremely sorry for the delay. I was completing other priority task, and this got delayed. Will not happen again.

**Comment by garrettmknight** [comment]
@shubham1206agra offer out to you!

**Comment by garrettmknight** [comment]
Dropping to weekly for @sobitneupane to request payment

**Comment by garrettmknight** [comment]
Not overdue. Still waiting for request for payment.

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [ ]  The PR that introduced the bug has been identified. Link to the PR:

This is an edge case and was probably existent from the inception of the feature. 
https://github.com/Expensify/App/pull/17119 and https://github.com/Expensify/App/pull/17445 are the PRs that worked on scrolling of the page.
> * [ ]  The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

N/A
> * [ ]  A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
> * [ ]  Determine if we should create a regression test for this bug.

Yes.
> * [ ]  If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Comment by sobitneupane** [comment]
**Regression Test Proposal**

1. Open Expensify App and logout if you are already logged in.
2. Click outside е-mail input field so "Please enter an email or phone number' is presented
3. Scroll down to the bottom of the page
4. Click on Create A New Account or Log in
5. Verify that Е-mail input field is focused.

Do we agree 👍 or 👎

**Comment by garrettmknight** [comment]
Done, closing

**Comment by JmillsExpensify** [comment]
$500 payment approved for @sobitneupane based on [this summary](https://github.com/Expensify/App/issues/32508#issuecomment-1896237115).
